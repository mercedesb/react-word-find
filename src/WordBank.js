import React, { Component } from 'react'
import PropTypes from 'prop-types'

export class WordBank extends Component {
  state = {
    newWord: ''
  }

  onChange = (event) => {
    this.setState({ newWord: event.target.value })
  }

  onAdd = (event) => {
    event.preventDefault()
    this.props.onAdd(this.state.newWord)
    this.setState({ newWord: '' })
  }

  render () {
    return (
      <div className='WordBank'>
        <div className='WordBank-container'>
          { this.props.words.map((word, i) =>
            <div className='WordBank-word' key={i}>{word}</div>
            )
          }
        </div>
        <form className='WordBank-addContainer'>
          <input 
            type='text' 
            onChange={this.onChange}
            value={this.state.newWord}
          />
          <button type='submit' className='Button Button--ghost' onClick={this.onAdd}>Add Word</button>
        </form>
      </div>
    )
  }
}

WordBank.propTypes = {
  words: PropTypes.array,
  onAdd: PropTypes.func
}

