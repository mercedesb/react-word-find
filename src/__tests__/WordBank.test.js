/* eslint-env jest */
import * as React from 'react'
import { shallow } from 'enzyme'
import { WordBank } from '../WordBank'

let subject 
let words
let onAdd

function loadSubject (props = {}) {
  words = ['hello', 'world']
  onAdd = jest.fn()
  subject = shallow(<WordBank
    words={words}
    onAdd={onAdd}
    {...props}
  />)
}

describe('#render', () => {
  it('renders properly', () => {
    loadSubject()
    expect(subject).toMatchSnapshot()
  })
})

describe('#onChange', () => {
  it('sets the state', () => {
    loadSubject()
    subject.instance().onChange({ target: { value: 'newword' } } )

    expect(subject.state('newWord')).toEqual('newword')
  })
})

describe('#onAdd', () => {
  it('calls the onAdd function from props', () => {
    loadSubject()
    subject.setState( { newWord: 'newword' } )
    subject.instance().onAdd()
    expect(onAdd).toHaveBeenCalledWith('newword')
  })

  it('sets the state', () => {
    loadSubject()
    subject.setState( { newWord: 'newword' } )
    subject.instance().onAdd()
    expect(subject.state('newWord')).toEqual('')
  })
})







