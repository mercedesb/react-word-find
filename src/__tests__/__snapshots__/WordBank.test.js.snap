// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WordBank
    onAdd={[Function]}
    words={
        Array [
            "hello",
            "world",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label>
          Word Bank
</label>,
        <div
          className="WordBank-container"
>
          <div
                    className="WordBank-word"
          >
                    hello
          </div>
          <div
                    className="WordBank-word"
          >
                    world
          </div>
</div>,
        <form
          className="WordBank-addContainer"
>
          <input
                    onChange={[Function]}
                    type="text"
                    value=""
          />
          <button
                    className="Button Button--ghost"
                    onClick={[Function]}
                    type="submit"
          >
                    Add Word
          </button>
</form>,
      ],
      "className": "WordBank",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Word Bank",
        },
        "ref": null,
        "rendered": "Word Bank",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="WordBank-word"
>
              hello
</div>,
            <div
              className="WordBank-word"
>
              world
</div>,
          ],
          "className": "WordBank-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": "hello",
              "className": "WordBank-word",
            },
            "ref": null,
            "rendered": "hello",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": "world",
              "className": "WordBank-word",
            },
            "ref": null,
            "rendered": "world",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              onChange={[Function]}
              type="text"
              value=""
/>,
            <button
              className="Button Button--ghost"
              onClick={[Function]}
              type="submit"
>
              Add Word
</button>,
          ],
          "className": "WordBank-addContainer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add Word",
              "className": "Button Button--ghost",
              "onClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": "Add Word",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
            Word Bank
</label>,
          <div
            className="WordBank-container"
>
            <div
                        className="WordBank-word"
            >
                        hello
            </div>
            <div
                        className="WordBank-word"
            >
                        world
            </div>
</div>,
          <form
            className="WordBank-addContainer"
>
            <input
                        onChange={[Function]}
                        type="text"
                        value=""
            />
            <button
                        className="Button Button--ghost"
                        onClick={[Function]}
                        type="submit"
            >
                        Add Word
            </button>
</form>,
        ],
        "className": "WordBank",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Word Bank",
          },
          "ref": null,
          "rendered": "Word Bank",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="WordBank-word"
>
                hello
</div>,
              <div
                className="WordBank-word"
>
                world
</div>,
            ],
            "className": "WordBank-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": "hello",
                "className": "WordBank-word",
              },
              "ref": null,
              "rendered": "hello",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": "world",
                "className": "WordBank-word",
              },
              "ref": null,
              "rendered": "world",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                onChange={[Function]}
                type="text"
                value=""
/>,
              <button
                className="Button Button--ghost"
                onClick={[Function]}
                type="submit"
>
                Add Word
</button>,
            ],
            "className": "WordBank-addContainer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add Word",
                "className": "Button Button--ghost",
                "onClick": [Function],
                "type": "submit",
              },
              "ref": null,
              "rendered": "Add Word",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
